bugsnag-plugin-android-anr/src/main/java/com/bugsnag/android/AnrPlugin.kt

```diff
@@ -16,12 +22,45 @@ internal class AnrPlugin : BugsnagPlugin {
     }
 
     private fun handleAnr(thread: Thread, client: Client) {
+        // generate a full report as soon as possible, then wait for extra process error info
         val errMsg = "Application did not respond to UI input"
         val exc = BugsnagException("ANR", errMsg, thread.stackTrace)
+        val error = Error.Builder(client.config, exc, client.sessionTracker, thread, true)
+            .severity(Severity.ERROR)
+            .severityReasonType(HandledState.REASON_ANR)
+            .build()
+
+        // wait for the ANR dialog to show, reporting once the information is available
+        collectAnrErrorDetails(client, error)
+    }
 
-        client.cacheAndNotify(
-            exc, Severity.ERROR, MetaData(),
-            HandledState.REASON_ANR, null, thread
-        )
+    private fun collectAnrErrorDetails(client: Client, error: Error) {
+        // TODO add timeouts, and better way of posting async runnable
+
+        Async.run(object : Runnable {
+            override fun run() {
+                try {
+                    Thread.sleep(INFO_POLL_THRESHOLD_MS)
+                    val anrDetails = collector.collectAnrDetails(client.appContext)
+
+                    if (anrDetails == null) {
+                        Logger.warn("Looping in attempt to capture ANR details")
+                        Async.run(this)
+                    } else {
+                        Logger.warn("Captured ANR details: $anrDetails")
+
+                        // TODO determine how to populate error report
+                        val metaData = error.metaData
+                        metaData.addToTab("ANR", "Message", anrDetails.shortMsg)
+                        metaData.addToTab("ANR", "Component", anrDetails.tag)
```

Comment:
Isn't this the same as the default context, since it shows the activity name? If not, is it a better context for what is there currently?

Reply:
I don't think this is the same as the default context. I believe if an ANR occurs in a `BroadcastReceiver` this value will be different - although we'll need to test that hypothesis.

Reply:
That would be pretty cool if so!

Reply:
This does not appear to be the case in my testing with a broadcastReceiver.